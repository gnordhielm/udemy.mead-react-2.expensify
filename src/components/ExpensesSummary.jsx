import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import {
  getVisibleExpenses,
  getExpensesTotal
} from '../selectors/expenses'

export const ExpensesSummary = ({
  expensesCount, expensesTotal
}) => (
    <p>
      Viewing {expensesCount} {expensesCount === 1 ? 'expense' : 'expenses'} totalling {numeral(expensesTotal).format('$0,0.00')}
    </p>
)

export default connect(({ expenses, filters }) => {
  const filteredExpenses = getVisibleExpenses(expenses, filters)

  return {
    expensesCount: filteredExpenses.length,
    expensesTotal: getExpensesTotal(filteredExpenses)
  }

})(ExpensesSummary)
