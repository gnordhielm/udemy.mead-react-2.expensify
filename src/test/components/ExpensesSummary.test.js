import 'react-dates/initialize'

import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary.jsx'

test('should render ExpensesSummary with no expenses', () => {

  const wrapper = shallow(<ExpensesSummary
    expenseCount={0}
    expensesTotal={0}
  />)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary with one expense', () => {

  const wrapper = shallow(<ExpensesSummary
    expenseCount={1}
    expensesTotal={100}
  />)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary with expenses', () => {
  const wrapper = shallow(<ExpensesSummary
    expenseCount={45}
    expensesTotal={12345}
  />)
  expect(wrapper).toMatchSnapshot()
})
