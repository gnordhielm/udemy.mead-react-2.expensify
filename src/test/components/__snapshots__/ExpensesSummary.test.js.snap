// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with expenses 1`] = `
<p>
  Viewing 
   
  expenses
   totalling 
  $12,345.00
</p>
`;

exports[`should render ExpensesSummary with no expenses 1`] = `
<p>
  Viewing 
   
  expenses
   totalling 
  $0.00
</p>
`;

exports[`should render ExpensesSummary with one expense 1`] = `
<p>
  Viewing 
   
  expenses
   totalling 
  $100.00
</p>
`;
